### PR TITLE
Ignore paths

### PR DESCRIPTION
**Pull request by jasonolmstead33**
Adding functionality to allow ignored paths in the HTTPTracer

### CODE REVIEW

File: http.go [hunk 1]
```diff
@@ -186,56 +187,65 @@ func (i *responseInterceptor) Header() http.Header {
 // TracedHTTPHandler returns a http.Handler that is traced as an opentracing.Span
 func TracedHTTPHandler(
 	h http.Handler,
+	ignoredPaths *regexp.Regexp,
```
Comment (andyday) [inline http.go]: actually we should use the interceptor for this.  you can add an ignore field to the SpanConfig and set that however you want in the interceptor.

**Review by andyday — CHANGES_REQUESTED** [review]

File: http.go [hunk 2]
```diff
@@ -189,53 +189,64 @@ func TracedHTTPHandler(
...
-			tracer: tracer,
-			span:   span,
-			ctx:    ctx.(core.SpanContext),
-			writer: w,
-		}
-
-		if parentCtx != nil {
-			ri.parentCtx = parentCtx.(core.SpanContext)
+		if ignored {
+			h.ServeHTTP(w, r)
```
Comment (andyday) [inline http.go]: add a return after this and then you can remove the else below and reduce nesting...

File: http.go [hunk 3]
```diff
@@ -189,53 +189,64 @@ func TracedHTTPHandler(
...
-			ext.Component("ctrace.TracedHttpHandler"),
-			ext.HTTPRemoteAddr(r.RemoteAddr),
-			ext.HTTPMethod(r.Method),
-			ext.HTTPUrl(r.URL.String()),
-			ext.HTTPUserAgent(r.UserAgent()),
+		ignored := false
+		for _, v := range interceptor {
+			config := v(r)
+			if config.IgnoredPaths != nil {
+				ignored = config.IgnoredPaths.MatchString(r.URL.String())
```
Comment (andyday) [inline http.go]: you can remove the ignored var and do

```js
if config.IgnoredPaths != null {
  if config.IgnoredPaths.MatchString(r.URL.String()) {
    h.ServeHTTP(w, r)
    return
  }
}
```

File: span_config.go [hunk 4]
```diff
@@ -9,6 +13,9 @@ type SpanConfig struct {
 	// OperationName is the custom operation name decided by interceptor
 	OperationName string
 
+	// IgnoredPaths are URL paths ignored by tracer (ex.. /health to be ignored as it is used by a load balancer)
+	IgnoredPaths *regexp.Regexp
```
Comment (andyday) [inline span_config.go]: Since this is a single expression i would change the name to `Ignore`

**Review by andyday — CHANGES_REQUESTED** [review]

**Comment by jasonolmstead33** [comment]
@andyday check it out now. I can close this and make the PR into HEAD if you like it.

**Comment by andyday** [comment]
looks good.  you can open the same pr into HEAD.

**Pull request merged**